src/Core/src/Platform/Windows/FrameworkElementExtensions.cs

```diff
@@ -324,5 +324,19 @@ internal static void SetApplicationResource(this FrameworkElement frameworkEleme
 					.ToPlatform()
 					.GetLocationRelativeTo(relativeTo.ToPlatform());
 		}
+
+		internal static void RefreshThemeResources(this FrameworkElement nativeView)
+		{
+			var previous = nativeView.RequestedTheme;
+
+			// Workaround for https://github.com/dotnet/maui/issues/7820
+			nativeView.RequestedTheme = nativeView.ActualTheme switch
+			{
+				ElementTheme.Dark => ElementTheme.Light,
+				_ => ElementTheme.Dark
+			};
+
+			nativeView.RequestedTheme = previous;
+		}
```

Comment:
This is the real "fix" here and is basically a workaround with a still-workaround-but-not-totally fix coming in the core Windows App SDK.